### PR TITLE
plat-sam: add SiP SMC for usb suspend

### PR DESCRIPTION
**Pull request by clementleger**
This series adds support for a SMC that allows to set suspend mode. This SMC will be used by the Linux driver when entering suspend.

### CODE REVIEW

File: core/arch/arm/plat-sam/sam_sfr.c [hunk 1]
```diff
@@ -25,3 +28,26 @@ vaddr_t sam_sfr_base(void)
 
 	return (vaddr_t)va;
 }
+
+static TEE_Result atmel_sfr_probe(const void *fdt, int node,
+				  const void *compat_data __unused)
+{
+	if (_fdt_get_status(fdt, node) != DT_STATUS_OK_SEC)
+		return TEE_ERROR_GENERIC;
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-sam/sam_sfr.c:49]: I guess this will lead an error reported in the log, is that the intent?

Comment (clementleger) [reply core/arch/arm/plat-sam/sam_sfr.c:49]: You are right, this was not intended, I always tested with a secure SFR node. I'll modify that.

Thanks

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by clementleger** [comment]
New version:
 - Add Jens Wiklander Acked-by

**Review by etienne-lms — COMMENTED** [review]
minor comments only

File: core/arch/arm/plat-sam/sam_sfr.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2017 Timesys Corporation.
+ * Copyright (C) 2021 Microchip
+ * All rights reserved.
+ */
+
+#include <mm/core_mmu.h>
+#include <mm/core_memprot.h>
```
[the remark this answers is not in the document]
Comment (clementleger) [reply core/arch/arm/plat-sam/sam_sfr.c:12]: Acked

File: core/arch/arm/plat-sam/sam_sfr.c [hunk 3]
```diff
@@ -5,12 +5,15 @@
...
 
+#include <kernel/dt.h>
+#include <matrix.h>
 #include <mm/core_mmu.h>
 #include <mm/core_memprot.h>
 #include <sama5d2.h>
-#include <sam_sfr.h>
 #include <types_ext.h>
 
+#include <sam_sfr.h>
```
[the remark this answers is not in the document]
Comment (clementleger) [reply core/arch/arm/plat-sam/sam_sfr.c:16]: Indeed.

**Comment by clementleger** [comment]
New version:
- Fix @etienne-lms  comments

File: core/arch/arm/plat-sam/sam_sfr.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-sam/sam_sfr.c:12]: not addressed.

File: core/arch/arm/plat-sam/main.c [hunk 4]
```diff
@@ -56,20 +56,6 @@ void console_init(void)
...
-	static void *va;
-
-	if (cpu_mmu_enabled()) {
-		if (!va)
-			va = phys_to_virt(SFR_BASE, MEM_AREA_IO_SEC, 1);
-		return (vaddr_t)va;
-	}
-	return SFR_BASE;
-}
-
```
Comment (etienne-lms) [inline core/arch/arm/plat-sam/main.c:72]: Maybe could remove `#include <sam_sfr.h>` at line 44.

File: core/arch/arm/plat-sam/sam_sfr.h [hunk 5]
```diff
@@ -7,6 +7,7 @@
 #define SAM_SFR_H
 
 #include <util.h>
+#include <stdbool.h>
```
Comment (etienne-lms) [inline core/arch/arm/plat-sam/sam_sfr.h:10]: swap the 2 above lines

**Comment by jforissier** [comment]
@clementleger this PR seems almost ready to be merged, could you please address the latest comments by @etienne-lms?

**Comment by clementleger** [comment]
> @clementleger this PR seems almost ready to be merged, could you please address the latest comments by @etienne-lms?

Yes, I'll do that today. Sorry, missed that one.

**Comment by clementleger** [comment]
New version:
 - Fixed @etienne-lms comments
 - Rename `SAMA5_SIP_SFR_SET_USB_SUSPEND` to `SAMA5_SMC_SIP_SFR_SET_USB_SUSPEND` to be more coherent

**Comment by jforissier** [comment]
Merged with Étienne's tag added. Thanks.

**Pull request closed without merging**